pkg/sql/schemachanger/scbuild/internal/scbuildstmt/helpers.go

```diff
@@ -364,7 +365,7 @@ func mustGetColumnIDFromColumnName(
 ) catid.ColumnID {
 	colID := getColumnIDFromColumnName(b, tableID, columnName)
 	if colID == 0 {
-		panic(errors.AssertionFailedf("cannot find column with name %v", columnName))
+		panic(colinfo.NewUndefinedColumnError(string(columnName)))
```

Comment:
since this is a "must get" function, i'm not sure what the contract is. it's unclear to me if the function contract is that it only should be called with a valid column ID, or if it should handle that case.

- if it should only be called with a valid column ID, then assertion error makes sense, and we should use a different function in cases when we are looking up a column ID that may not exist
- if it should handle unknown column IDs, maybe we shouldn't name it `must...`

Reply:
This wrapper is generally bad since none of these cases should hit assertion failures (the names are from user input). So, I have modified everything to use: getColumnIDFromColumnName, and have a flag that indicates if the resolution is required (leading to a user-friendly error).